src/components/Schedule/SubmitButton.tsx

```diff
@@ -0,0 +1,96 @@
+import { gql, useMutation } from "@apollo/client";
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  Button,
+  Flex,
+  useToast,
+} from "@chakra-ui/react";
+import { useEffect } from "react";
+import { UpdateScheduleInput } from "../../stores/schedule";
+
+const UPDATE = gql`
+  mutation update($input: UpdateScheduleInput!) {
+    updateSchedule(input: $input) {
+      message
+      success
+    }
+  }
+`;
+
+interface SubmitButtonProps {
+  handleSubmit: () => UpdateScheduleInput;
+  active: boolean;
+  setActive: (active: boolean) => void;
+}
+
+export const SubmitButton = (props: SubmitButtonProps) => {
+  const { handleSubmit, active, setActive } = props;
+
+  const [generate, { data, loading, error }] = useMutation(UPDATE);
```

Comment:
Kind of too late for this but the coupling of state here between this component and the parent is a bit scuffed. Normally you would either do all the state management in this component (don't pass in `active` and `handleSubmit`) or do everything in the parent component (do the mutation inside of `handleSubmit` instead of in this component). Not worth changing anything if this works just a comment.

Reply:
yeah definitely scuffed lol